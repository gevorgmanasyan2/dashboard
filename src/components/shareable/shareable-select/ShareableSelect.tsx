import React, { FC } from 'react'
import './shareableSelect.scss'
import { Select } from 'antd'
import { SharableProps } from '../models'

const ShareableSelect: FC<SharableProps> = ({ label, value, children, onChange }) => {
  return (
    <div className="shareableSelect__container">
      <label htmlFor="">{label}</label>
      <Select
        defaultValue={value}
        onChange={onChange}
        className="shareableSelect__container--select"
      >
        {children}
      </Select>
    </div>
  )
}

export default ShareableSelect
