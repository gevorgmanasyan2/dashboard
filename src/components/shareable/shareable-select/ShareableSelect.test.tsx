/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import ShareableSelect from './ShareableSelect'

it('should render ShareableSelect component', () => {
  const mockItem = {
    label: 'Status',
  }
  expect(shallow(<ShareableSelect {...mockItem} />)).toMatchSnapshot()
})
