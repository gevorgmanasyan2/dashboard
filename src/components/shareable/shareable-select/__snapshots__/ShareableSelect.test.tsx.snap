// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShareableSelect component 1`] = `
<div
  className="shareableSelect__container"
>
  <label
    htmlFor=""
  >
    Status
  </label>
  <Select
    bordered={true}
    choiceTransitionName=""
    className="shareableSelect__container--select"
    transitionName="slide-up"
  />
</div>
`;
